generic3g/specs/VectorBracketClassAspect.F90

```diff
@@ -0,0 +1,336 @@
+         aspect_ids = [ &
+              CLASS_ASPECT_ID, &
+              GEOM_ASPECT_ID &
+           ]
+      end if
+
+      ! Othrerwise doing geom regrid first is a performance improveent.
+      aspect_ids = [ &
+           GEOM_ASPECT_ID, &
+           CLASS_ASPECT_ID &
+           ]
+
+      _RETURN(_SUCCESS)
+      _UNUSED_DUMMY(goal_aspects)
+   end function get_aspect_order
+
+   subroutine create(this, handle, rc)
+      class(VectorBracketClassAspect), intent(inout) :: this
+      integer, optional, intent(in) :: handle(:) 
+     integer, optional, intent(out) :: rc
+
+     integer :: status
+     type(ESMF_Info) :: info
+
+      this%payload = MAPL_FieldBundleCreate(fieldBundleType=FIELDBUNDLETYPE_VECTOR_BRACKET, _RC)
+      _RETURN_UNLESS(present(handle))
+
+      call ESMF_InfoGetFromHost(this%payload, info, _RC)
+      call FieldBundleInfoSetInternal(info, spec_handle=handle, allocation_status=STATEITEM_ALLOCATION_CREATED, bracket_updated=.true.,  _RC)
+      call MAPL_FieldBundleSet(this%payload, allocation_status=STATEITEM_ALLOCATION_CREATED, _RC)
+
+      _RETURN(_SUCCESS)
+   end subroutine create
+
+   subroutine activate(this, rc)
+      class(VectorBracketClassAspect), intent(inout) :: this
+      integer, optional, intent(out) :: rc
+
+      integer :: status
+
+      call MAPL_FieldBundleSet(this%payload, allocation_status=STATEITEM_ALLOCATION_ACTIVE, _RC)
+
+      _RETURN(_SUCCESS)
+   end subroutine activate
+
+   ! Tile / Grid   X  or X, Y
+   subroutine allocate(this, other_aspects, rc)
+      class(VectorBracketClassAspect), intent(inout) :: this
+      type(AspectMap), intent(in) :: other_aspects
+      integer, optional, intent(out) :: rc
+
+      integer :: status
+      integer :: i
+      type(FieldClassAspect) :: tmp
+
+      associate (n => this%bracket_size)
+        
+        do i = 1, n
+           tmp = this%field_aspect
+           call tmp%create(_RC)
+           call tmp%allocate(other_aspects, _RC)
+           call tmp%add_to_bundle(this%payload, _RC)
+        end do
+      end associate
+
+      _RETURN(_SUCCESS)
+
+   contains
+
+      function int_to_string(i) result(s)
+         character(:), allocatable :: s
+         integer, intent(in) :: i
+         character(len=20) :: buffer
+         write(buffer, '(i0)') i
+         s = trim(buffer)
+      end function int_to_string
+
+   end subroutine allocate
+
+
+  subroutine destroy(this, rc)
+      class(VectorBracketClassAspect), intent(inout) :: this
+      integer, optional, intent(out) :: rc
+
+      integer :: status
+      integer :: i
+      type(ESMF_Field), allocatable :: fieldList(:)
+
+      call MAPL_FieldBundleGet(this%payload, fieldList=fieldList, _RC)
+      do i = 1, size(fieldList)
+         call ESMF_FieldDestroy(fieldList(i), noGarbage=.true., _RC)
+      end do
+      call ESMF_FieldBundleDestroy(this%payload, noGarbage=.true., _RC)
+
+      _RETURN(_SUCCESS)
+   end subroutine destroy
```

Comment:
Note that @darianboggs is in the process of providing a MAPL service for this.   Let's try to remember to go back and clean this up when his interface is available.